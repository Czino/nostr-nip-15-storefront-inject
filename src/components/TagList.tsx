import { NDKTag } from '@nostr-dev-kit/ndk'

type Props = { tags: NDKTag[] }

export const TagList = ({ tags }: Props) => (
    <div className="flex flex-row gap-1 flex-wrap">
        {tags
            .filter((t) => t[0] === 't')
            .map(([, tag]) => (
                <a
                    key={tag}
                    className="px-2 py-1 text-xs bg-light-1 text-dark-1 rounded cursor-pointer"
                    href={`/shop?tags=${tag}`}
                >
                    {tag}
                </a>
            ))}
    </div>
)
