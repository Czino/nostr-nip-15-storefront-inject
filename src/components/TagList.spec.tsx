import { render } from '@testing-library/react'
import { TagList } from './TagList'

describe('TagList', () => {
    it('renders tags correctly', () => {
        const { asFragment } = render(
            <TagList
                tags={[
                    ['t', 'cat'],
                    ['t', 'balaclava'],
                    ['s', 'unrelated tag'],
                ]}
            />,
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
