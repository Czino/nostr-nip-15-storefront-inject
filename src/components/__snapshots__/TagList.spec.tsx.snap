// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList renders tags correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row gap-1 flex-wrap"
  >
    <a
      class="px-2 py-1 text-xs bg-light-1 text-dark-1 rounded cursor-pointer"
      href="/shop?tags=cat"
    >
      cat
    </a>
    <a
      class="px-2 py-1 text-xs bg-light-1 text-dark-1 rounded cursor-pointer"
      href="/shop?tags=balaclava"
    >
      balaclava
    </a>
  </div>
</DocumentFragment>
`;
